codegen-test/model/rest-xml-extras.smithy

```diff
@@ -61,7 +61,7 @@ structure CreateFooInput {}
         </PrimitiveIntDocument>
        """,
        bodyMediaType: "application/xml",
-       params: { value: 1 }
+       params: { value: 1, requiredValue: 0 }
```

Comment:
The test description is: "Primitive ints should not be serialized when they are unset", so adding `requiredValue` to the params seems to break what is being tested. Are these tests still valid with Smithy IDL 2?

Reply:
[Primitives](https://github.com/awslabs/smithy/pull/1286) are now just the type with a default value backing it.

And default values are [always serialized.](https://awslabs.github.io/smithy/2.0/spec/type-refinement-traits.html#default-value-serialization)

Reply:
I think the model for this test needs to be revised because it seems like the test isn't testing what it intended to test anymore. It seems like the [`requiredValue`](https://github.com/awslabs/smithy-rs/blob/2be10ed9dd9348336b239ecd5b1467ad221d36c0/codegen-test/model/rest-xml-extras.smithy#L75) should get a default set on it rather than providing a value for it in the test case.

Reply:
The change I made was on the basis of understanding that old behavior around primitives and `@required` were kind of a wild west -- apparently not just for nullability, but also serialization. `@default` in 2.0 is where the behavior starts to be defined and predicted so I upgraded the test model to 2.0 and used that trait instead.

I don't believe there is anything further to be tested.